content/doc/book/pipeline/scaling-pipeline.adoc

```diff
@@ -0,0 +1,88 @@
+
+== Will Higher-Performance Durability Settings Help Me?
+* Yes, if you are running complex Pipelines or Pipelines with many steps (more than several hundred).
+* Yes, if your Pipeline stores large files or complex data to variables in the script and then runs steps.  This sounds oddly specific but happens more than you'd expect.
+* Yes, if your Jenkins instance uses NFS, magnetic storage, runs many Pipelines at once, or shows high iowait.
+* No, if your Pipelines spend almost all their time waiting for a few shell/batch scripts to finish.  This ISN'T a magic "go fast" button for everything!
+* No, if Pipelines are writing massive amounts of data to logs (logging is unchanged).
+* No, if you are not using Pipelines, or your system is loaded down by other factors.
+* No, if you don't enable higher-performance modes for pipelines.
+
+== What Am I Giving Up With This Durability Setting "Trade-Off?"
+
+*Stability of Jenkins ITSELF is not changed regardless of this setting* - it only applies to Pipelines.  The worst-case behavior for Pipelines reverts to something like FreeStyle builds -- running Pipelines that cannot persist transient data may not be able to resume or be displayed in Blue Ocean/Stage View/etc, but will show logs.  This impacts _only_ running Pipelines and only when Jenkins is shut down abruptly and not gracefully before they get to complete.
+
+A *"graceful" shutdown* is where Jenkins goes through a full shutdown process, such as visiting http://[jenkins-server]/exit,  or using normal service shutdown scripts (if Jenkins is healthy).  Sending a SIGTERM/SIGINT to Jenkins will trigger a graceful shutdown.  
+
+A *"dirty" shutdown* is when Jenkins does not get to do normal shutdown processes. This can occur if the process is forcibly terminated.  The most common causes are using SIGKILL to terminate the Jenkins process or killing the container/VM running Jenkins.  Simply stopping or pausing the container/VM will not cause this, as long as the Jenkins process is able to resume. 
+A dirty shutdown can also happen due to catastrophic operating system failures, including the Linux OOMKiller attacking the Jenkins java process to free memory.
+
+All but the most durable setting ("MAX_SURVIVABILITY") currently avoid atomic writes -- what this means is that if the Operating System fails, data that is buffered for writing to disk will not be flushed, it will be lost.  This is quite rare, but can happen as a result of container or virtualization operations that halt the operating system or disconnect storage.  Usually this data is flushed pretty quickly to disk, so the window for data loss is brief.  On Linux this flush-to-disk can be forced by running 'sync'.
+
+== Requirements To Use Durability Settings
+
+* Jenkins LTS 2.73+ or higher (or a weekly 2.62+)
+* Current Pipeline Plugin versions from near the end of January 2018 for all the following:
```

Comment:
If they're released already, we're in mid January.

Better yet, specify exact minimum versions or this is going to make @rtyler cry.

Reply:
@daniel-beck Hoping to release today (having some docs coming out / about to come out is release requirement).  My hope was to include more specific versions once released... but don't tempt me to omit it. :trollface: